go/internal/benchmark/benchmark_test.go

```diff
@@ -0,0 +1,201 @@
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"berty.tech/berty/v2/go/internal/testutil"
+	"berty.tech/berty/v2/go/pkg/bertymessenger"
+	bertymessengertesting "berty.tech/berty/v2/go/pkg/bertymessenger/testing"
+	"berty.tech/berty/v2/go/pkg/messengertypes"
+)
+
+func b64DecodeBytes(s string) ([]byte, error) {
+	return base64.RawURLEncoding.DecodeString(s)
+}
+
+func BenchmarkScenario(b *testing.B) {
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Second)
+	defer cancel()
+	logger, cleanup := testutil.Logger(b)
+	defer cleanup()
+	accountsAmount := 3
+	clients, _, cleanup := bertymessengertesting.NonMockedTestingInfra(b, accountsAmount)
+	defer cleanup()
+
+	// create nodes
+	var creator *bertymessenger.TestingAccount
+	{
+		creator = bertymessenger.NewTestingAccount(ctx, b, clients[0], nil, logger)
+		defer creator.Close()
+		creator.SetName(b, "Creator")
+	}
+	joiners := make([]*bertymessenger.TestingAccount, accountsAmount-1)
+	{
+		for i := 0; i < accountsAmount-1; i++ {
+			joiners[i] = bertymessenger.NewTestingAccount(ctx, b, clients[i+1], nil, logger)
+			defer joiners[i].Close()
+			joiners[i].SetName(b, "Joiner #"+strconv.Itoa(i))
+		}
+	}
+
+	// Force delay
+	time.Sleep(time.Second)
+
+	// creator creates a new conversation
+	convName := "Ze Conv"
+	createdConv, err := creator.GetClient().ConversationCreate(ctx, &messengertypes.ConversationCreate_Request{DisplayName: convName})
+	require.NoError(b, err)
+
+	// get conv link
+	gpk := createdConv.GetPublicKey()
+	gpkb, err := b64DecodeBytes(gpk)
+	require.NoError(b, err)
+	sbg, err := creator.GetClient().ShareableBertyGroup(ctx, &messengertypes.ShareableBertyGroup_Request{GroupPK: gpkb, GroupName: convName})
+	require.NoError(b, err)
+
+	// joiners join the conversation
+	for _, joiner := range joiners {
+		ret, err := joiner.GetClient().ConversationJoin(ctx, &messengertypes.ConversationJoin_Request{Link: sbg.GetWebURL()})
+		require.NoError(b, err)
+		require.Empty(b, ret)
+	}
+
+	// wait for events propagation
+	time.Sleep(2 * time.Second)
+
+	// open streams and drain lists on all nodes
+	accounts := append([]*bertymessenger.TestingAccount{creator}, joiners...)
+	for _, account := range accounts {
+		account.DrainInitEvents(b)
+		account.Close()
+	}
+
+	// no more event
+	{
+		event := creator.TryNextEvent(b, 200*time.Millisecond)
+		require.Nil(b, event)
+		for _, joiner := range joiners {
+			event = joiner.TryNextEvent(b, 200*time.Millisecond)
+			require.Nil(b, event)
+		}
+	}
+
+	// verify members in each account
+	for _, account := range accounts {
+		accountConv := account.GetConversation(b, gpk)
+		require.Equal(b, gpk, accountConv.GetPublicKey())
+		require.Equal(b, convName, accountConv.GetDisplayName())
+		mpk := account.GetConversation(b, gpk).GetAccountMemberPublicKey()
+		require.NotEmpty(b, mpk)
+		//displayName := account.GetAccount().GetDisplayName()
```

Comment:
clean?

Reply:
This is a copied version of `TestPeersCreateJoinConversationNonMocked` in `bertymessenger`, I'll remove those commented lines